accepted/00101-health-check-tool.md

```diff
@@ -0,0 +1,282 @@
+
+* Implementing alerts and recommendations involves:
+    * Defining LogQL Queries: Writing LogQL queries that match the relevant conditions.
+    * Configuring Alert Rules: Using the Loki Ruler to define alert rules based on LogQL queries. These rules specify the conditions under which alerts should be triggered and the severity levels.
+    * Setting Up Notification Channels: Configuring the notification channels to receive alerts.
+    * Recommendation Logic: For recommendations, the alerting mechanism can be extended with additional logic to suggest actions.
+
+
+## Metrics
+
+* Server status and resource utilization
+
+* Database health and integrity
+
+* Status of relevant Uyuni services
+
+
+## Saline vs Health Check Tool
+
+We want to differentiate conceptually between "Saline" and the "Health Check Tool". In this regard, these are two different components with different purposes.
+
+### Saline: Salt state appliement monitoring to be integrated to the product.
+Saline exposes Salt state metrics to Prometheus. It is a tool meant to attach to Salt Master sockets and analyze and extract Salt state/jobs metrics from a live and running Uyuni server.
+
+Saline setup is done is two steps:
+
+1. Install `saline` RPM in the Uyuni server. This provides "setup" script and new "Formulas with Forms".
+2. Run `saline-setup run` to configure and enable the `salined` service to attach to the Salt Master.
+3. In the "Formulas with Forms" UI, new formulas will appear: "Saline Prometheus" and "Saline Grafana". These can be used to automatically configure your Prometheus and Grafana instances to get the metrics and dashboards from Saline.
+
+Saline already provides an easy way to integrate with the current Monitoring Stack (Formulas with Forms).
+
+The `salined` service must be able to attach to the Salt Master. At the moment this is done by connecting to Salt Master IPC sockets, which means `salined` needs to live in the same system/container than the Salt Master. There already an ongoing research to make Salt Master to use TCP instead of IPC sockets, that would help us at the time we want to split the different components of Uyuni server into individual containers.
+
+The Saline engine that is running on the Uyuni server, can be also used to feed a future UI with a live view of running actions. It is a potential component that we may want eventually integrate as part of the default Uyuni server stack.
+
+
+### Health Check Tool: A tool you can run on a live server or use in a disconnected setup (via supportconfig)
+In this case, this tool is mean not only to provide a picture of the current health status of the Uyuni server, but also to help engineers to debug disconnected setups via "supportconfig". In this regard, the tool cannot always rely on existing Prometheus and Grafana instances, but it should be able to deploy its own instances to visualize the data on disconnected environments.
```

Comment:
Can you add an explanation of the terms "disconnected setup" and "disconnected environment"? I don't know what exactly you mean with those terms.